### DISCUSSION TITLE
Prisma client seems to want both the relation as well as the scalar - and neither work

### DISCUSSION THREAD

**Discussion by FunctionDJ** [prisma/prisma#18959] [Q&A] [answered]
Relevant parts of my schema:
```
model Match {
  map   WWMap  @relation(fields: [mapId], references: [id])
  mapId String
}

model WWMap {
  id              String   @id @default(cuid())
  matches         Match[]
  @@map("Map")
}
```

This passes type checking but will error and say that "mapId" isn't specified:
```ts
      prisma.match.create({
        data: {
          map: {
            connect: {
              id: input.mapId,
            },
          },
        },
      });
```
So i tried also giving it the mapId as a property of `data` but then it says `Unknown arg mapId` and i'm just very confused.

**Comment by FunctionDJ** [comment, marked as answer]
The error was that i wrote a different column (not shown in code above) that has the type Json but i wrote `null` to the DB and then Prisma client just gave me this really weird error.

**Reply by nurul3101** [reply]
Thanks for getting back and letting is know that this is resolved 🙌
